sylt-compiler/src/typechecker.rs

```diff
@@ -1133,8 +1136,20 @@ impl TypeChecker {
 
             ExpressionKind::Blob { blob, fields } => {
                 let blob_ty = self.type_assignable(ctx, blob)?;
+
+                let mut errors = Vec::new();
+
                 let (blob_name, blob_fields) = match self.find_type(blob_ty) {
                     Type::Blob(name, fields) => (name, fields),
+                    Type::Newblob(name, fields, _) => {
+                        errors.push(type_error!(
+                            self,
+                            span,
+                            TypeError::NewblobInstance { name: name.clone() }
+                        ));
+
+                        (name, fields)
+                    }
                     _ => unreachable!(),
```

Comment:
This is a lie! I'll make an issue of it.

```
A :: enum
end

start :: fn do
    A {}
end
```
thread 'main' panicked at 'internal error: entered unreachable code', sylt-compiler/src/typechecker.rs:1153:26